source/common/network/proxy_protocol.cc

```diff
@@ -143,11 +139,11 @@ bool ProxyProtocol::ActiveConnection::readLine(int fd, std::string& s) {
       }
     }
 
+    // Read the data upto and including the line feed, if available, but not past it.
+    // This should never fail, as search_index_ - buf_off_ <= nread, so we're asking
+    // only for bytes we have already seen.
     nread = recv(fd, buf_ + buf_off_, search_index_ - buf_off_, 0);
-
-    if (nread < 1) {
-      throw EnvoyException("failed to read proxy protocol");
-    }
+    ASSERT(size_t(nread) == search_index_ - buf_off_);
```

Comment:
The outer loop `while (buf_off_ < MAX_PROXY_PROTO_LEN)` terminates in this case, as but_off_ becomes MAX_PROXY_PROTO_LEN, and the exception after the loop kicks in.

I did not know that ASSERTs are elided in opt builds, but it should not matter, as the assert should not fire in the first place. If you feel necessary, I could transform the assert to an if statement instead.